### PR TITLE
heap output for MNPR, MScalar, and MScalarSUN

### PR DESCRIPTION
**Pull request by krox**
here are the changes as discussed. AFAIK thats all done.

### CODE REVIEW

**Comment by krox** [comment]
uh, wait. this is wrong. only meant to PR my changes

File: Hadrons/Modules/MScalar/ChargedProp.cpp [hunk 1]
```diff
@@ -39,23 +40,18 @@ TChargedProp::TChargedProp(const std::string name)
 {}
 
 // dependencies/products ///////////////////////////////////////////////////////
-std::vector<std::string> TChargedProp::getInput(void)
+std::vector<std::string> TChargedProp::getInput()
 {
-    std::vector<std::string> in = {par().source, par().emField};
-    
-    return in;
+    return {par().source, par().emField};
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/ChargedProp.cpp:45]: We should keep `void` in the argument signature and keep `in` as a separate `std::vector<std::string>` for consistency with the rest of Hadrons here.

File: Hadrons/Modules/MScalar/ChargedProp.cpp [hunk 2]
```diff
@@ -39,23 +40,18 @@ TChargedProp::TChargedProp(const std::string name)
...
 }
 
-std::vector<std::string> TChargedProp::getOutput(void)
+std::vector<std::string> TChargedProp::getOutput()
 {
-    std::vector<std::string> out = {getName(), getName()+"_0", getName()+"_Q",
-                                    getName()+"_Sun", getName()+"_Tad"};
-    
-    return out;
+    return {getName(), getName()+"_0", getName()+"_Q", getName()+"_Sun", getName()+"_Tad", getName()+"_projections"};
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/ChargedProp.cpp:50]: We should keep `void` in the argument signature and keep `out` as a separate `std::vector<std::string>` for consistency with the rest of Hadrons here.

File: Hadrons/Modules/MScalar/ChargedProp.cpp [hunk 3]
```diff
@@ -39,23 +40,18 @@ TChargedProp::TChargedProp(const std::string name)
...
 }
 
 // setup ///////////////////////////////////////////////////////////////////////
-void TChargedProp::setup(void)
+void TChargedProp::setup()
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/ChargedProp.cpp]: Keep void in argument sig for consistency with the rest of Hadrons

File: Hadrons/Modules/MScalar/ChargedProp.cpp [hunk 4]
```diff
@@ -85,14 +81,15 @@ void TChargedProp::setup(void)
...
+    envCreate(HadronsSerializable, getName()+"_projections", 1, 0);
     envTmpLat(ScalarField, "buf");
     envTmpLat(ScalarField, "result");
     envTmpLat(ScalarField, "Amu");
     envCache(FFT, fftName_, 1, env().getGrid());
 }
 
 // execution ///////////////////////////////////////////////////////////////////
-void TChargedProp::execute(void)
+void TChargedProp::execute()
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/ChargedProp.cpp]: Keep void in argument sig for consistency with the rest of Hadrons

File: Hadrons/Modules/MScalar/FreeProp.cpp [hunk 5]
```diff
@@ -39,32 +40,29 @@ TFreeProp::TFreeProp(const std::string name)
 {}
 
 // dependencies/products ///////////////////////////////////////////////////////
-std::vector<std::string> TFreeProp::getInput(void)
+std::vector<std::string> TFreeProp::getInput()
 {
-    std::vector<std::string> in = {par().source};
-    
-    return in;
+    return {par().source};
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/FreeProp.cpp:45]: We should keep `void` in the argument signature and keep `in` as a separate `std::vector<std::string>` for consistency with the rest of Hadrons here.

File: Hadrons/Modules/MScalar/FreeProp.cpp [hunk 6]
```diff
@@ -39,32 +40,29 @@ TFreeProp::TFreeProp(const std::string name)
...
 }
 
-std::vector<std::string> TFreeProp::getOutput(void)
+std::vector<std::string> TFreeProp::getOutput()
 {
-    std::vector<std::string> out = {getName()};
-    
-    return out;
+    return {getName(), getName()+"_sliceSum"};
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/FreeProp.cpp:50]: We should keep `void` in the argument signature and keep `out` as a separate `std::vector<std::string>` for consistency with the rest of Hadrons here.

File: Hadrons/Modules/MScalar/FreeProp.cpp [hunk 7]
```diff
@@ -39,32 +40,29 @@ TFreeProp::TFreeProp(const std::string name)
...
 }
 
 // setup ///////////////////////////////////////////////////////////////////////
-void TFreeProp::setup(void)
+void TFreeProp::setup()
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/FreeProp.cpp]: Keep void in the argument sig for consistency with Hadrons

File: Hadrons/Modules/MScalar/FreeProp.cpp [hunk 8]
```diff
@@ -39,32 +40,29 @@ TFreeProp::TFreeProp(const std::string name)
...
     
     freePropDone_ = env().hasCreatedObject(freeMomPropName_);
     envCacheLat(ScalarField, freeMomPropName_);
     envCreateLat(ScalarField, getName());
+    envCreate(HadronsSerializable, getName()+"_sliceSum", 1, 0);
 }
 
 // execution ///////////////////////////////////////////////////////////////////
-void TFreeProp::execute(void)
+void TFreeProp::execute()
```
Comment (RChrHill) [inline Hadrons/Modules/MScalar/FreeProp.cpp]: Keep void in the argument sig for consistency with the rest of Hadrons

File: Hadrons/Modules/MScalarSUN/Div.hpp [hunk 9]
```diff
@@ -93,28 +94,27 @@ TDiv<SImpl>::TDiv(const std::string name)
 
 // dependencies/products ///////////////////////////////////////////////////////
 template <typename SImpl>
-std::vector<std::string> TDiv<SImpl>::getInput(void)
+std::vector<std::string> TDiv<SImpl>::getInput()
```
Comment (RChrHill) [inline Hadrons/Modules/MScalarSUN/Div.hpp]: Keep void in the argument sig for consistency with the rest of Hadrons

File: Hadrons/Modules/MScalarSUN/Div.hpp [hunk 10]
```diff
@@ -93,28 +94,27 @@ TDiv<SImpl>::TDiv(const std::string name)
...
 
 template <typename SImpl>
-std::vector<std::string> TDiv<SImpl>::getOutput(void)
+std::vector<std::string> TDiv<SImpl>::getOutput()
 {
-    std::vector<std::string> out = {getName()};
-
-    return out;
+    return {getName(), getName()+"_sum"};
 }
```
Comment (RChrHill) [inline Hadrons/Modules/MScalarSUN/Div.hpp:106]: We should keep `void` in the argument signature and keep `out` as a separate `std::vector<std::string>` for consistency with the rest of Hadrons here.

File: Hadrons/Modules/MScalarSUN/Div.hpp [hunk 11]
```diff
@@ -93,28 +94,27 @@ TDiv<SImpl>::TDiv(const std::string name)
...
 
 // setup ///////////////////////////////////////////////////////////////////////
 template <typename SImpl>
-void TDiv<SImpl>::setup(void)
+void TDiv<SImpl>::setup()
```
Comment (RChrHill) [inline Hadrons/Modules/MScalarSUN/Div.hpp:110]: Keep void in the argument sig for consistency with the rest of Hadrons

File: Hadrons/Modules/MScalarSUN/TwoPoint.hpp [hunk 12]
```diff
@@ -119,9 +120,7 @@ std::vector<std::string> TTwoPoint<SImpl>::getInput(void)
 template <typename SImpl>
 std::vector<std::string> TTwoPoint<SImpl>::getOutput(void)
 {
-    std::vector<std::string> out = {};
-
-    return out;
+    return {getName()};
```
Comment (RChrHill) [inline Hadrons/Modules/MScalarSUN/TwoPoint.hpp:123]: We should keep `out` as a separate `std::vector<std::string>` for consistency with the rest of Hadrons here.

File: Hadrons/Modules/MScalarSUN/TwoPointNPR.hpp [hunk 13]
```diff
@@ -105,9 +106,7 @@ std::vector<std::string> TTwoPointNPR<SImpl>::getInput(void)
 template <typename SImpl>
 std::vector<std::string> TTwoPointNPR<SImpl>::getOutput(void)
 {
-    std::vector<std::string> out;
-    
-    return out;
+    return {getName()};
```
Comment (RChrHill) [inline Hadrons/Modules/MScalarSUN/TwoPointNPR.hpp:109]: We should keep `out` as a separate `std::vector<std::string>` for consistency with the rest of Hadrons here.

**Review by RChrHill — CHANGES_REQUESTED** [review]
Looks good overall, just a few minor stylistic points to revert in order to keep overall consistency within Hadrons.

**Comment by krox** [comment]
I re-added the `(void)` parameters. These have exactly no effect in (any version of) C++, so you might be right for stylistic consistency. But I cant bring myself to write `std::vector<std::string> in = ...; return in` . That just looks silly to me to needlessly repeat the type name. And in some (older) versions of C++ it can even be less efficient. (Definitely not relevant here of course. Just a matter of principle).

I know I am a bit of a style snob here, I am sorry. This discussion is not really worth my or your time. I will not object to any changes you feel like making. Thanks for your work.

**Comment by RChrHill** [comment]
I'll merge the PR and clear up any requests for further style changes from higher up myself. Thanks for your work!

**Pull request merged**